### PR TITLE
docs: refine readme and example codes

### PR DESCRIPTION
**Pull request by lkevinzc**
- Try to make the example code explanation clearer
- Refine diffuser example codes
- Fix typos
- Point example reference link to docs, since we have illustrations there

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,9 +48,16 @@ pip install -U mosec
 
 ## Usage
 
+We demonstrate how Mosec can easily host a pre-trained stable diffusion model as a service. You need to install [diffusers](https://github.com/huggingface/diffusers) and [transformers](https://github.com/huggingface/transformers) as prerequisites:
```
Comment (lkevinzc) [inline README.md:51]:
```suggestion
We demonstrate how Mosec can help you easily host a pre-trained stable diffusion model as a service. You need to install [diffusers](https://github.com/huggingface/diffusers) and [transformers](https://github.com/huggingface/transformers) as prerequisites:
```

File: README.md [hunk 2]
```diff
@@ -65,19 +72,19 @@ from mosec.mixin import MsgpackMixin
...
+Then, we **build an API** for clients to query a text prompt and obtain an image based on the [stable-diffusion-v1-5 model](https://huggingface.co/runwayml/stable-diffusion-v1-5). To achieve that, we simply inherit the `MsgpackMixin` and `Worker` class, then initialize the model in the `__init__` method and override the `forward` method. Note that the input `data` of `forward` method is by default a JSON-decoded object, but `MsgpackMixin` will override it to use [msgpack](https://msgpack.org/index.html) for both request and response data. In this example, the `forward` method will wishfully receive a _list_ of string, e.g., `['a cute cat playing with a red ball', 'a man sitting in front of a computer', ...]`, aggregated from different clients for _batch inference_, improving the system throughput. Also note that the returned objects will be encoded by the msgpack protocol as well, due to the use of `MsgpackMixin`.
```
Comment (kemingy) [inline README.md:75]: I think `Note that ...` can be a separate sentence with an explanation about why we need msgpack.

File: README.md [hunk 3]
```diff
@@ -86,12 +93,10 @@ class StableDiffusion(MsgpackMixin, Worker):
             dummy_file = BytesIO()
             img.save(dummy_file, format="JPEG")
             images.append(dummy_file.getbuffer())
-        # need to return the same number of images in the same request order
-        # `len(data) == len(images)`
         return images
 ```
 
-Finally, we append the worker to the server to construct a *single-stage* workflow, and specify the number of processes we want it to run in parallel. Then run the server.
+Finally, we append the worker to the server to construct a *single-stage* workflow (multiple stages can be [pipelined](https://en.wikipedia.org/wiki/Pipeline_(computing)) to further boost the throughput, see [this example](https://mosecorg.github.io/mosec/example/pytorch/#computer-vision)), and specify the number of processes we want it to run in parallel (`num=1`), and the maximum batch size (`max_batch_size=16`, the maximum number of requests dynamic batching will accumulate before timeout).
```
Comment (kemingy) [inline README.md:99]: The last `timeout` may confuse users. I think it's better to clarify the relation with `--wait` here.

File: README.md [hunk 4]
```diff
@@ -65,19 +71,19 @@ from mosec.mixin import MsgpackMixin
...
+Then, we **build an API** for clients to query a text prompt and obtain an image based on the [stable-diffusion-v1-5 model](https://huggingface.co/runwayml/stable-diffusion-v1-5). To achieve that, we simply inherit the `Worker` and `MsgpackMixin` class, then initialize the model in the `__init__` method and override the `forward` method. By default, the input `data` of the `forward` method should be a JSON-decoded object. Using `MsgpackMixin` will override it to use the faster and smaller [msgpack](https://msgpack.org/index.html) for both request and response data. In this example, the `forward` method will wishfully receive a _list_ of string, e.g., `['a cute cat playing with a red ball', 'a man sitting in front of a computer', ...]`, aggregated from different clients for _batch inference_, improving the system throughput. Also note that the returned objects will be encoded by the msgpack protocol as well, due to the use of `MsgpackMixin`.
```
Comment (kemingy) [inline README.md:74]: This paragraph is a little bit long. We can split it into two.

Also, we use msgpack because we need to return an image, which is bytes. But JSON doesn't support bytes unless you encode it with base64 (which makes it longer).

Comment (kemingy) [reply README.md:74]: Maybe we can use the GitHub Flavor README note block like:
```
> **Note**
>
> Long Descriptions here.
```

> **Note**
>
> Long Descriptions here.

Comment (lkevinzc) [reply README.md:74]: I feel this "Note" sign kind of breaks the content, since this para describes the second step of implementing a service. I will try to split it into bullet points such that it's clearer

File: README.md [hunk 5]
```diff
@@ -65,19 +71,36 @@ from mosec.mixin import MsgpackMixin
...
+
+
+> **Note**
+>
+> (a) In this example we return an image in the binary format, which JSON does not support (unless encoded with base64 that makes it longer). Hence, msgpack suits our need better. If we do not inherit `MsgpackMixin`, JSON will be used by default. In other words, the protocol of the service request/response can either be msgpack or JSON.
+>
+> (b) Warm-up usually helps to pre-allocate GPU memory in advance. If the warm-up example is specified, the service will only be ready after the example is forwarded through the handler. However, if no example is given, the first request's latency is expected to be longer. The `example` should be set as a single item or a tuple depending on what `forward` expects to receive. Moreover, in the case where you want to warm up with multiple different examples, you may set `multi_examples` (demo [here](https://mosecorg.github.io/mosec/example/jax/)).
```
Comment (kemingy) [inline README.md:87]:
```suggestion
> (b) Warm-up usually helps to allocate GPU memory in advance. If the warm-up example is specified, the service will only be ready after the example is forwarded through the handler. However, if no example is given, the first request's latency is expected to be longer. The `example` should be set as a single item or a tuple depending on what `forward` expects to receive. Moreover, in the case where you want to warm up with multiple different examples, you may set `multi_examples` (demo [here](https://mosecorg.github.io/mosec/example/jax/)).
```
Duplicate: `pre` and `in advance`.

**Review by kemingy — APPROVED** [review]
